jobclient/python/cookclient/__init__.py

```diff
@@ -356,19 +356,19 @@ def _apply_jobspec_defaults(jobspec: dict) -> dict:
         returned.
         """
         jobspec = copy.deepcopy(jobspec)
-        if 'uuid' not in jobspec:
+        if jobspec.get('uuid', None) is None:
             jobspec['uuid'] = str(util.make_temporal_uuid())
-        if 'cpus' not in jobspec:
+        if jobspec.get('cpus', None) is None:
             jobspec['cpus'] = 1.0
-        if 'mem' not in jobspec:
+        if jobspec.get('mem', None) is None:
             jobspec['mem'] = 128.0
-        if 'max-retries' not in jobspec:
+        if jobspec.get('max-retries', None) is None:
             jobspec['max-retries'] = 1
-        if 'max-runtime' not in jobspec:
+        if jobspec.get('max-runtime', None) is None:
             jobspec['max-runtime'] = timedelta(days=1)
-        if 'name' not in jobspec:
+        if jobspec.get('name', None) is None:
             jobspec['name'] = f'{getpass.getuser()}-job'
-        if 'application' not in jobspec:
+        if jobspec.get('application', None) is None:
```

Comment:
Maybe we should define a function, e.g. `is_field_set` or similar, that takes in a dictionary and field and returns `d.get(field, None) is not None` and use that for all of these? I think you could use it below as well.